lms/static/scripts/frontend_apps/components/AppRoot.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'preact/hooks';
+
+import type { ConfigObject } from '../config';
+import { Config } from '../config';
+import type { ServiceMap } from '../services';
+import { Services } from '../services';
+import BasicLTILaunchApp from './BasicLTILaunchApp';
+import ErrorDialogApp from './ErrorDialogApp';
+import FilePickerApp from './FilePickerApp';
+import OAuth2RedirectErrorApp from './OAuth2RedirectErrorApp';
+
+export type AppRootProps = {
+  /** Initial route and configuration for the frontend, read from the HTML page. */
+  initialConfig: ConfigObject;
+  services: ServiceMap;
+};
+
+/**
+ * The root component for the LMS frontend.
+ */
+export default function AppRoot({ initialConfig, services }: AppRootProps) {
+  const [config] = useState(initialConfig);
```

Comment:
To support route transitions in future, we'll need to be able to augment the configuration after the app loads. This is not yet implemented.